vyper/semantics/types/subscriptable.py

```diff
@@ -263,6 +263,11 @@ def compare_type(self, other):
             return False
         if self.length < other.length:
             return False
+
+        # handle class objects for empty arrays - see `types.PRIMITIVE_TYPES`
```

Comment:
this is kind of a very odd check, since it means something is being passed to the dynarray constructor which is not really a type. we don't want to encourage that. instead, i think we should fix at the location where the incorrect `value_type` is introduced in the first place: https://github.com/vyperlang/vyper/blob/7b9cf67a055580929c259398b3829adc6edd57f6/vyper/semantics/analysis/utils.py#L334-L340

here we should check if the type is actually a type class, and insert type_class.any() into the list instead of the type class.